test/sqllogictest/list.slt

```diff
@@ -1243,81 +1319,347 @@ SELECT length(LIST['1','2']::text)
 query error Cannot call function length\(string list\): arguments cannot be implicitly cast to any implementation's parameters; try providing explicit casts
 SELECT length(LIST['1','2'])
 
-# 🔬🔬 list_prepend
+# 🔬🔬 text to list
 
-# 🔬🔬🔬 element + list
+# 🔬🔬🔬 text to int list
+
+query I
+SELECT * FROM unnest('{1,2,3}'::int list)
+----
+1
+2
+3
+
+query T rowsort
+SELECT * FROM unnest('{{1,2},{3}}'::int list list)
+----
+{1,2}
+{3}
+
+query T rowsort
+SELECT * FROM unnest('{{{1},{2}},{{3}}}'::int list list list);
+----
+{{1},{2}}
+{{3}}
 
 query T
-SELECT list_prepend(1, LIST[2, 3])
+SELECT '{NULL}'::int list;
 ----
-{1,2,3}
+{NULL}
 
-# 🔬🔬🔬 list + list list (element + list)
+query T
+SELECT '{}'::int list
+----
+{}
+
+# 🔬🔬🔬 text to text list
 
 query T
-SELECT list_prepend(LIST[1], LIST[[2], [3]])
+SELECT '{"a"," b","c ", d,e }'::text list
 ----
-{{1},{2},{3}}
+{a," b","c ",d,e}
 
-# 🔬🔬🔬 NULL elements
+query T
+SELECT '{{"a"," b","c "},{ d,e }}'::text list list
+----
+{{a," b","c "},{d,e}}
 
-# 🔬🔬🔬🔬 element + list
+# NULL-esque string handling
+query TI rowsort
+SELECT unnest, length(unnest) from  unnest('{NULL,nUlL,"NULL"}'::text list) ORDER BY length
+----
+NULL  4
+NULL  NULL
+NULL  NULL
+
+# Test some escape values
+query TI rowsort
+SELECT unnest, length(unnest) from  unnest('{"a,b","\\","a\\b\"c\\d\""}'::text list) ORDER BY length
+----
+\  1
+a,b  3
+a\b"c\d"  8
 
 query T
-SELECT list_prepend(NULL, LIST[1])
+SELECT '{NULL}'::text list;
 ----
-{NULL,1}
+{NULL}
 
 query T
-SELECT list_prepend(NULL::INT, LIST[1])
+SELECT '{}'::text list
 ----
-{NULL,1}
+{}
+
+# 🔬🔬🔬🔬 Quote escapes
 
+# Preserves leading and trailing spaces within escape, but not outside of escape
```

Comment:
These tests were very helpful in understanding the `match` logic in `parse_list`!